lib/server.js

```diff
@@ -204,15 +209,25 @@ function handleRequest(opts, req, resp) {
         }),
         log: null,
         reqId: req.headers['x-request-id'],
-        metrics: opts.metrics
+        metrics: opts.metrics,
+        startTime: Date.now()
     };
     reqOpts.log = reqOpts.logger.log.bind(reqOpts.logger);
 
-    // Simplistic count of requests from public & private networks
+    /**
+     * We use separate metric trees for requests from private networks, so
+     * that we can separately study performance for those different use cases.
+     * We implement this by passing a specialized prefixed metric producer to
+     * the handler. All derived metrics associated with a private / update
+     * request are thus recorded in .internal or .internal_update prefixed
+     * metric trees.
+     */
     if (/^(?:::ffff:)?(?:10|127)\./.test(remoteAddr)) {
-        reqOpts.metrics.increment('requests.private');
-    } else {
-        reqOpts.metrics.increment('requests.public');
+        if (req.headers['cache-control'] && /no-cache/i.test(req.headers['cache-control'])) {
```

Comment:
I'm thinking about correctness here - it is not inconceivable to have an internal client that issues a `no-cache` request. Granted, this will not happen on regular basis, but still...